### PR TITLE
Update providers structure

### PR DESCRIPTION
**Pull request by tylerhutcherson**
Update `redisvl` `Providers` to `Vectorizers` >>
```
redisvl/
  vectorize/
    __init__.py
    base.py
    text/
        __init__.py
        huggingface.py
        openai.py
    # image/
```

Also creates classes for `HFTextVectorizer` and `OpenAITextVectorizer`.

### CODE REVIEW

File: redisvl/llmcache/semantic.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
 
 from redisvl.index import SearchIndex
 from redisvl.llmcache.base import BaseLLMCache
-from redisvl.providers import HuggingfaceProvider
-from redisvl.providers.base import BaseProvider
+from redisvl.vectorize import HuggingfaceVectorizer
```
Comment (Spartee) [inline redisvl/llmcache/semantic.py]: ``redisvl.vectorize.text``?

File: README.md [hunk 2]
```diff
@@ -42,11 +42,12 @@ RedisVL has a host of powerful features designed to streamline your vector datab
...
 
-4. **Semantic Caching**: ``LLMCache`` is a semantic caching interface built directly into RedisVL. It allows for the caching of generated output from LLM models like GPT-3 and others. As semantic search is used to check the cache, a threshold can be set to determine if the cached result is relevant enough to be returned. If not, the model is called and the result is cached for future use. This can increase the QPS and reduce the cost of using LLM models.
+4. **Powerful Abstractions**
+    - **Semantic Caching**: `LLMCache` is a semantic caching interface built directly into RedisVL. It allows for the caching of generated output from LLMs like GPT-3 and others. As semantic search is used to check the cache, a threshold can be set to determine if the cached result is relevant enough to be returned. If not, the model is called and the result is cached for future use. This can increase the QPS and reduce the cost of using LLM models in production.
```
Comment (Spartee) [inline README.md:50]: mean to leave the bullet in here?

**Review by Spartee — COMMENTED** [review]
Minor comments, but looks like it's failing looking for the API key. IT's a secret so that should work.

File: README.md (hunk 2 above)
Comment (tylerhutcherson) [reply README.md:50]: I did in an effort to break out this section more and highlight that more abstractions and improvements are coming

File: redisvl/llmcache/semantic.py (hunk 1 above)
Comment (tylerhutcherson) [reply redisvl/llmcache/semantic.py]: yeah, probably best - will update

**Comment by tylerhutcherson** [comment]
> Minor comments, but looks like it's failing looking for the API key. IT's a secret so that should work.

Interesting. Can we confirm the str used? The tests pass locally for me with `OPENAI_API_KEY` set personally.

**Comment by tylerhutcherson** [comment]
> > Minor comments, but looks like it's failing looking for the API key. IT's a secret so that should work.
> 
> Interesting. Can we confirm the str used? The tests pass locally for me with `OPENAI_API_KEY` set personally.

Resolved!

**Pull request merged**